### ISSUE TITLE
Unnecessary suggestions to change active keyboard

### ISSUE THREAD

**Issue by NeatNit** [keymapperorg/KeyMapper#1063]
Feel free to change the title, I couldn't really phrase it well.

Take a look at these 2 key mappings (sorry for large image size):

![image](https://user-images.githubusercontent.com/2400026/178004268-3b46f671-a3f0-46a4-bef7-aec78b1c761f.png)

The app highlights the actions and warns that they can't work because Key Mapper Basic Input Method (KMBIM for short) is not active. At the time of taking this screenshot, Samsung Keyboard is the active input method.

However, these triggers have a constraint that prevents them from being triggered to begin with if KMBIM isn't active - meaning I am explicitly already aware of this limitation and took it into account. I think these warnings shouldn't be shown in this case.

I'll go even a step further - maybe Key Mapper should suggest adding this constraint *as an alternative fix* for this warning.

Another alternative fix might be to add an action to switch to KMBIM before the input action, but in my testing this doesn't seem to work without adding a delay, so I would further suggest adding a "wait until completed" option to the Switch Keyboard action which produces no delay if the keyboard is already selected, but some delay if not - ideally the delay would simply be a wait for some "ready" signal from KMBIM if that's possible. Edit: This is very related to #797 (which I only found because I just cloned this project and noticed an old branch related to it)

**Comment by NeatNit** [comment]
I am starting work on a pull request implementing only the first thing I said - don't show a warning if the constraint takes care of it.

The other related suggestions are a design decision that you should make yourself so I don't want to step on your toes - besides, they are a lot more complex to implement, so I probably wouldn't do a good job at it :P

I should warn you, I have a bad habit of not finishing what I start, so I might disappear without any pull request :( if you don't hear from me in a while you can just assume I've done that and implement that yourself if you decide to do that.

**Comment by jambl3r** [comment]
Thanks for your issue. I just talked with the developer and we had a big discussion on design choices regarding this. We've decided it's better to just warn the user in all cases. If the warnings are hidden in some cases like this it is inevitable someone will make a key map like this and not understand why it's working... because there is no warning. So for now I'll close this issue, but feel free to make a pull request anyway and I'll have another look
